### ISSUE TITLE
BasePlot marker should allow no label defined

### ISSUE THREAD

**Issue by HealthyPear** [steveberardi/starplot#115]
**Describe the bug**

If I do not specify a label to a marker I get

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[132], line 52
     42 p.circle(
     43     center = (ra.to_value("hourangle"), dec.to_value("deg")),
     44     radius_degrees = radius,
   (...)
     49     )
     50 )
---> 52 p.marker(
     53         ra=ra.to_value("hourangle"),
     54         dec=dec.to_value("deg"),
     55         # label="",
     56         legend_label="search center",
     57         style={
     58             "marker": {
     59                 "size": 5,
     60                 "symbol": "plus",
     61                 "fill": "full",
     62                 "color": "black",
     63                 "alpha": 1,
     64                 "zorder": 100,
     65             },
     66         },
     67 )

File [~/Software/miniforge3/envs/astro/lib/python3.12/site-packages/starplot/styles/helpers.py:81](http://localhost:8889/lab/workspaces/auto-l/tree/~/Software/miniforge3/envs/astro/lib/python3.12/site-packages/starplot/styles/helpers.py#line=80), in use_style.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
     77 elif style is None and style_attr is not None:
     78     # if no style overrides and there's a base style, then just pass the base style
     79     kwargs["style"] = getattr(args[0].style, style_attr, None)
---> 81 return func(*args, **kwargs)

TypeError: BasePlot.marker() missing 1 required positional argument: 'label'
```

**To Reproduce**

see above

**Expected behavior**

I should be able to not specify a label

**Environment Details (please complete the following information):**
- Starplot version = 0.12.3
- Python version = 3.12.7
- Matplotlib version = 3.84
- OS: macOS-14.7.1-arm64-arm-64bit

**Comment by steveberardi** [comment]
Agreed, this has been annoying me too 😂 

I'll add this to the roadmap 👍 easy to fix

**Comment by steveberardi** [comment]
I just released this change in v0.12.4, thanks again!
